client/js/team.ts

```diff
@@ -0,0 +1,52 @@
+const joinTeamButton = document.getElementById('joinTeam');
```

Comment:
Also, you can avoid the non-null assertion operators (`!`) below by putting `as HTMLInputElement` at the end of this line. We should keep those non-null operators' use to a minimum because they can swallow errors and problems with the code.